app/src/main/java/it/niedermann/owncloud/notes/model/LocalAccount.java

```diff
@@ -59,6 +77,54 @@ public void setModified(long modified) {
         this.modified = modified;
     }
 
+    public ApiVersion getPreferredApiVersion() {
+        return preferredApiVersion;
+    }
+
+    /**
+     * @param availableApiVersions <code>["0.2", "1.0", ...]</code>
+     */
+    public void setPreferredApiVersion(@Nullable String availableApiVersions) {
+        // TODO move this logic to NotesClient?
+        try {
+            if(availableApiVersions == null) {
+                this.preferredApiVersion = null;
+                return;
+            }
+            JSONArray versionsArray = new JSONArray(availableApiVersions);
+            Collection<ApiVersion> supportedApiVersions = new ArrayList<>(versionsArray.length());
+            for (int i = 0; i < versionsArray.length(); i++) {
+                ApiVersion parsedApiVersion = ApiVersion.of(versionsArray.getString(i));
+                for (ApiVersion temp : NotesClient.SUPPORTED_API_VERSIONS) {
+                    if (temp.compareTo(parsedApiVersion) == 0) {
+                        supportedApiVersions.add(parsedApiVersion);
```

Comment:
`compareTo` is implemented with an `equal`-like functionality. This is not the intended way to get the supported API versions. You should only check the major version of the API.

Example: The server provides "1.4, 2.1". Then, you chose "1.4", because you only support major version "1" but not "2". Due to backward/forward compatibility, your implementation for "1.0" will work on the "1.4" API endpoint.

Example 2: The server provides "0.2, 1.4" and your app has implemented the API for "0.2" and "1.7". Then, you should still chose "1.4" (since you support major version "1"). Requests will still work, but you must be aware, that the result may miss some fields that were not defined in "1.4" but in "1.7".

(API `0.2` is a special case, since the major version of `0.2` is `0.2` and not `0`.)

Reply:
`compareTo` now only compares the `major`-version of the API.